src/core/namespace.c

```diff
@@ -349,8 +349,9 @@ static int make_read_only(BindMount *m) {
                 r = bind_remount_recursive(m->path, true);
         else if (IN_SET(m->mode, READWRITE, PRIVATE_TMP, PRIVATE_VAR_TMP, PRIVATE_DEV)) {
                 r = bind_remount_recursive(m->path, false);
-                if (r == 0 && m->mode == PRIVATE_DEV) /* can be readonly but the submounts can't*/
-                        r = mount(NULL, m->path, NULL, MS_REMOUNT|DEV_MOUNT_OPTIONS|MS_RDONLY, NULL);
+                /* Private /dev can be readonly but the submounts can't */
+                if (r == 0 && m->mode == PRIVATE_DEV && mount(NULL, m->path, NULL, MS_REMOUNT|DEV_MOUNT_OPTIONS|MS_RDONLY, NULL) < 0)
+                        r = -errno;
```

Comment:
i am not a fan of embedded the actual function that it's supposed to be called here in the if check. That just makes this harder to read as it mixes up the condition check when to run this with the operaton what to do. I'd much rather see this written as:

```c
if (r >= 0 && m->mode == PRIVATE_DEV) {
        if (mount(...) < 0)
                r = -errno;
}
```

Even though this is a bit longer...